projects/RabbitMQ.Client/RabbitMQ.Client.csproj

```diff
@@ -1,7 +1,7 @@
-﻿<Project Sdk="Microsoft.NET.Sdk">
+<Project Sdk="Microsoft.NET.Sdk">
 
   <PropertyGroup>
-    <TargetFrameworks>netcoreapp3.1;netstandard2.0</TargetFrameworks>
+    <TargetFramework>netstandard2.0</TargetFramework>
```

Comment:
@danielmarbach based on what I've read and what you've pointed out, this seems to be the _simplest_ way to ensure that `RabbitMQ.Client` will work on the platforms you point out in [this comment](https://github.com/rabbitmq/rabbitmq-dotnet-client/pull/1189#issuecomment-1090439791).

If that's not the case then I really don't understand the relationships between different framework targets 😆

Reply:
Define simplest. It is the simplest way to get a broad range of platform of which many are a bit of a dead end like mono for example.

But that definition of simplest doesn't take into account that you can't multi target anymore and you need to do everything with shim packages. By having the possibility of multi targeting you can do platform specific optimizations and features which opens many interesting avenues to evolve the library.

The important part though is to define what is necessary based on your user base and then find the right mix for the future of the client.

Reply:
> But that definition of simplest doesn't take into account that you can't multi target anymore and you need to do everything with shim packages.

That went right over my head. My naive understanding is that by targeting `netstandard2.0` pretty much any application targeting a version compatible `netstandard2.0` can consume this library.

Reply:
For what it's worth we have had Xamarin related questions for this library (cc @michaelklishin since he may remember others).

Reply:
Yes that is the consumption part. That is true. But from a library perspective you can only write code against the netstandard2.0 compatible APIs. Good example is System.Memory. By default buffering is not in the netstandard APIs. So you pull in the System.Memory package. That package is only extended to a certain point and you might get stuck there in what you can benefit from as a library author. When you multi target you can pull on those shims for example only for net framework version. You can write code that is compatible with the feature intersection of the two target frameworks as some sort of a shared code path. Yet if you want to for example use some larest and greatest feature that comes with NET 6 you can leverage that by ifdef things. So users using that output assembly will be using the more optimized version. If you would only target netstandard2.0 you can't do that because you are only producing outputs for that common subset of APIs.

Of course you could also multi target two even more TFMs if you can deal with the maintenance burden that comes with it.

We have chosen to drop support for netstandard2.0 because it is a dead end in terms of evolution. Your mileage might vary though.

Some OSS libs dual target as a compromise netstandard2.0 and netstandard2.1 which is kind of a tradeoff too given netstandard 2.1 never really took off and the whole netstandard movement is a blast from the past

https://devblogs.microsoft.com/dotnet/the-future-of-net-standard/

Reply:
If you look at Immos comment

> My expectation is that widely used libraries will end up multi-targeting for both .NET Standard 2.0 and .NET 5: supporting .NET Standard 2.0 gives you the most reach while supporting .NET 5 ensures you can leverage the latest platform features for customers that are already on .NET 5.

And combine this with net5 is non Lts and out of support soon plus all the problems mentioned in that post about netstandard caveats net472 and Net6 is a good choice. Of course you can go with netstandard2.0 and Net6 too but then you also expose yourself with weird things like net462  not working really and the other problems in the blog post. That being said maybe for this lib these tradeoffs are OK. I can't know since I'm not sure I can represent a significant user base. Just providing input